### PR TITLE
docs: add snapshot feature doc

### LINKED ISSUE #152: add snapshot feature doc
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

**Describe alternatives you've considered**

Currently snapshot feature is already merged in v0.3.0, while need a doc about this
https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/snapshot

could create a doc under `snapshot.doc` under:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/docs

**Additional context**

/assign @ZeroMagic

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
add snapshot feature doc

**Which issue(s) this PR fixes**:

Fixes #152 

**Special notes for your reviewer**:

**Release note**:
```
add snapshot feature doc
```

### CODE REVIEW

File: deploy/example/snapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+outfile
+```
+
+## Create a Snapshot of the Source PVC
+
+```console
+kubectl apply -f $GOPATH/src/github.com/kubernetes-sigs/azuredisk-csi-driver/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml
+kubectl apply -f $GOPATH/src/github.com/kubernetes-sigs/azuredisk-csi-driver/deploy/example/snapshot/azuredisk-volume-snapshot.yaml
+```
+
```
Comment (feiskyer) [inline deploy/example/snapshot/README.md]: also add some statements on how to check the creation or restore status?

Comment (ZeroMagic) [reply deploy/example/snapshot/README.md]: done.

**Comment by andyzhangx** [comment]
/hold
Let's hold this PR a little since there is issue when verify on snapshot feature.

**Comment by andyzhangx** [comment]
/lgtm
/approve
/hold cancel
let's merge first

**Pull request merged**